lib/grape/dsl/inside_route.rb

```diff
@@ -91,19 +91,25 @@ def error!(message, status = nil, headers = nil)
       # @param url [String] The url to be redirect.
       # @param options [Hash] The options used when redirect.
       #                       :permanent, default false.
+      #                       :body, default a short message including the URL.
       def redirect(url, options = {})
-        merged_options = options.reverse_merge(permanent: false)
-        if merged_options[:permanent]
+        permanent = options.fetch(:permanent, false)
+        body_message = options.fetch(:body, nil)
+        if permanent
           status 301
+          body_message ||= "This resource has been moved permanently to #{url}"
```

Comment:
I don't think we need a validator, if someone wants to send something as
text in another language we shouldn't impose English rules :)

On Tue, Nov 10, 2015 at 10:16 AM, Tyler Dooling <notifications@github.com>
wrote:

> In lib/grape/dsl/inside_route.rb
> <https://github.com/ruby-grape/grape/pull/1194#discussion_r44417636>:
>
> >            status 301
> > +          body_message ||= "This resource has been moved permanently to #{url}"
>
> No worries - good point.
>
> Should we add a validator to the optional message override to ensure it
> adheres grammatically or just open up a separate issue for that? [image:
> :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ruby-grape/grape/pull/1194/files#r44417636>.
>



-- 

dB. | Moscow - Geneva - Seattle - New York
code.dblock.org - @dblockdotorg <http://twitter.com/#!/dblockdotorg> -
artsy.net - github/dblock <https://github.com/dblock>